### PR TITLE
Testing add validation tests to getInfo tests

### PR DESCRIPTION
**Pull request by omarahmed1111**
Testing PR for [UR PR](https://github.com/oneapi-src/unified-runtime/pull/1346)

After correcting hip program info returned for program device to return context device rather than the binary associated device that have fixed the kernel fusion cooperative kernels e2e test.

### CODE REVIEW

File: sycl/test-e2e/KernelFusion/cooperative_kernel.cpp [hunk 1]
```diff
@@ -3,11 +3,6 @@
 
 // Test cooperative kernels are not fused
 
-// This test currently fails on AMD HIP due to an unresolved problem when
-// submitting a kernel with the use_root_sync property to a command group with
-// fusion enabled.
-// XFAIL: hip_amd
-
```
Comment (omarahmed1111) [inline sycl/test-e2e/KernelFusion/cooperative_kernel.cpp:10]: This test is not failing anymore after changing hip program info returned for program device to return context device and not the binary associated device.

**Review by victor-eds — CHANGES_REQUESTED** [review]
Changes LGTM. Can we have an explanation of why the test is passing in the commit message?

**Comment by omarahmed1111** [comment]
> Changes LGTM. Can we have an explanation of why the test is passing in the commit message?

Done. Thanks for pointing that!

**Comment by victor-eds** [comment]
Correction:

>After correcting hip program info returned for program device to return context device rather than the binary associated device that have fixed the **kernel fusion** cooperative kernels e2e test.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 2]
```diff
@@ -94,14 +94,8 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  # commit 717791bfc636a1cd69f95f09fea15fcd70a9fb23
-  # Merge: 601062ba f8c4facd
-  # Author: aarongreig <aaron.greig@codeplay.com>
-  # Date:   Fri Apr 19 10:35:14 2024 +0100
-  #    Merge pull request #1517 from nrspruit/fix_l0_coverity_sync
-  #    [L0] Store LastCommandEvent before unlock during queue sync
-  set(UNIFIED_RUNTIME_TAG 717791bfc636a1cd69f95f09fea15fcd70a9fb23)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/omarahmed1111/unified-runtime.git")
+  set(UNIFIED_RUNTIME_TAG cbd581fbe4182ac443aa17edf2214cd05369a110)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:98]:
```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit a6cdcf4591779a925c5f017a19c26ea171797edf
  # Merge: 54829f00 cbd581fb
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Mon Apr 29 10:39:11 2024 +0100
  #     Merge pull request #1346 from omarahmed1111/ensure-getinfo-cts-tests-report-same-dependency-creation-object
  #     Add output validation checks for getInfo tests and a small USM spec fix
  set(UNIFIED_RUNTIME_TAG cbd581fbe4182ac443aa17edf2214cd05369a110)
```

**Review by kbenzie — COMMENTED** [review]
The UR PR has merged, please update the UR repo/tag varaibles as suggested.

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**